GoogleUtilities/AppDelegateSwizzler/Private/GULAppDelegateSwizzler.h

```diff
@@ -46,10 +46,43 @@ typedef NSString *const GULAppDelegateInterceptorID;
 /** This method ensures that the original app delegate has been proxied. Call this before
  *  registering your interceptor. This method is safe to call multiple times (but it only proxies
  *  the app delegate once).
+ *
+ *  This method doesn't proxy APNS related methods:
+ *  @code
+ *    - application:didRegisterForRemoteNotificationsWithDeviceToken:
+ *    - application:didFailToRegisterForRemoteNotificationsWithError:
+ *    - application:didReceiveRemoteNotification:fetchCompletionHandler:
+ *    - application:didReceiveRemoteNotification:
+ *  @endcode
+ *
+ *  To proxy these methods use +[GULAppDelegateSwizzler
+ * proxyOriginalDelegateRemoteNotificationMethods]. The methods have to be proxied separately to
+ * avoid potential warnings from Apple review about missing Push Notification Entitlement (e.g.
+ * https://github.com/firebase/firebase-ios-sdk/issues/2807)
+ *  @see proxyOriginalDelegateRemoteNotificationMethods
  */
 + (void)proxyOriginalDelegate NS_EXTENSION_UNAVAILABLE(
     "App delegate proxy doesn't support extensions.");
 
+/** This method ensures that the original app delegate has been proxied including APNS related
+ *  methods. Call this before registering your interceptor. This method is safe to call multiple
+ *  times (but it only proxies the app delegate once) or
+ *  after +[GULAppDelegateSwizzler proxyOriginalDelegate]
+ *
+ *  This method calls +[GULAppDelegateSwizzler proxyOriginalDelegate] under the hood.
```

Comment:
This is not obvious to the developers so the name seems confusing.
I prefer change the first method to `proxyOriginalDelegateWithoutRemoteNotificationMethods` and the second to `proxyOriginalDelegate`

Reply:
Sounds reasonable to me. 
@paulb777 @ryanwilson @tejasd What do you think?

Reply:
SGTM

Reply:
Renamed